### PR TITLE
Allows selecting all/none email preferences.

### LINKED ISSUE #2707: Add "select all" checkbox to email preferences
See designs on #2510

### PR DESCRIPTION
**Pull request by justinlittman**
closes #2707

## How was this change tested? 🤨

⚡ ⚠ If this change involves consuming from or writing to another service (or shared file system), ***run [integration test create_object_h2_spec.rb](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test manually in [stage|qa] environment, in addition to specs. ⚡

cypress

Note that deviation from mock-ups was @amyehodge approved.

### CODE REVIEW

**Comment by justinlittman** [comment]
image

File: app/javascript/controllers/checkboxes_controller.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { Controller } from "stimulus"
+
+export default class extends Controller {
+  static targets = ["checkbox"]
+
+  connect() {
+    console.log("Connected")
```
Comment (jcoyne) [inline app/javascript/controllers/checkboxes_controller.js:7]: Cruft.

**Review by jcoyne — CHANGES_REQUESTED** [review]
This doesn't seem to follow the UI as sketched out by @astridu.  Is there a reason to deviate from those designs?

**Comment by amyehodge** [comment]
@jcoyne @justinlittman and I talked about how there isn't a clear path for the user here when only some of the boxes are checked. Should we be displaying the "select all" or the "deselect all" option? This allows the user to use either of those options at any time and seemed a good solution.

File: app/javascript/controllers/checkboxes_controller.js (hunk 1 above)
Comment (justinlittman) [reply app/javascript/controllers/checkboxes_controller.js]: Removed.

**Review by jcoyne — APPROVED** [review]

**Pull request merged**